### DISCUSSION TITLE
左手キーボードの特定のキーが反応しない

### DISCUSSION THREAD

**Discussion by m-tky** [Yowkees/keyball#562] [Q&A]
![248109](https://github.com/Yowkees/keyball/assets/150902821/51e3f548-4a6a-4ea9-8569-5816b5be89c3)

赤で囲った部分の部分のキーが反応しません。

まず、症状が起こる前の状態についてです。
スイッチが反応しなくなってしまい、キースイッチを刺し直すと反応するようになり、しばらく立つと反応しなくなり、、というのをしばらく繰り返していました。その際に通電確認を行うと、うまく反応しているようでした。

そこで、一度ハンダ吸い取り線を使ってソケットを取り外し、別のものに付け替えました。
写真の図の赤で囲った部分が動かないのですが、ソケットの両端は、スイッチを押した際に通電しているようでした。
また、緑の線の部分（配線を確認して、つながっていると思われる部分）との通電も確認できました。

ソケットを取り外す際にはんだごてを当てすぎたのか、それともはんだが足りないのか、何が原因なのかがわかりません。
お手数をおかけしますがよろしくお願いします。

**Comment by Ryo624** [comment]
ご連絡ありがとうございます。

ひとつのキーの不具合なので回路や基板、Pro Microが原因である可能性は低いですね。

キースイッチを別のものに変えてみても反応はありませんでしょうか。
一度試していただけると幸いです。

**Reply by m-tky** [reply]
ご返信ありがとうございます。
他のキーが使えることは確認できているため、試しにそのキーと交換してみたのですが、やはりつかないようです。
また、反応しないキーについても、そこのスイッチを別の場所にはめたら反応しました。
どうやらスイッチの問題ではないようです。

**Reply by Ryo624** [reply]
ご対応いただきましてありがとうございます。
キースイッチ自体は正常に動作するということで了解しました。

通電の確認ですが写真に示した青い丸同士の箇所も確認していただいてもよろしいでしょうか。
![324914258-51e3f548-4a6a-4ea9-8569-5816b5be89c3](https://github.com/Yowkees/keyball/assets/143698349/702e20f6-6098-4004-bbdf-b7afa105d817)

**Reply by m-tky** [reply]
ありがとうございます。
先程試しましたが、問題なく通電するようです。

**Reply by Ryo624** [reply]
ご対応ありがとうございます。

キーソケットの左側の端子部分のはんだがかなり多いように見えますのでキーソケットの中にはんだが入り込んでしまってショートしているかもしれませんね。

予備のキーソケットがありましたらお手数をおかけし大変申し訳ありませんがもう一度キーソケットのはんだづけをお願いしてもよろしいでしょうか。
はんだの量には気を付けてはんだづけを行ってください。

**Reply by m-tky** [reply]
![2024-04-27-13-38-58-763~3.jpg](https://github.com/Yowkees/keyball/assets/150902821/a4b4b9d7-b525-4061-a347-0a667fcf7be1)

キーソケットを取り替えてみたのですが、まだ反応しないようです。
正確に言うと、キースイッチ（取り付ける時に何度かだけ反応したのですがその後は反応しなくなってしまいました。

赤のポイントは、キースイッチを押したときだけ通電しました。
また、黄、緑、青のポイントも、問題なく通電していました。

初心者であるため、お手数をおかけしてしまい申し訳ありません。
よろしくお願いします。



![2024-04-27-13-38-58-763~2.jpg](https://github.com/Yowkees/keyball/assets/150902821/46648e4d-53dd-4fb5-9727-4568bd1db88a)

**Reply by Ryo624** [reply]
黄色の線で示していただいたものが対応しているダイオードですが下記リンクのようにテスターでチェックしてみるとどうなりますでしょうか。

https://zenn.dev/halt/articles/20231029_keyball44#%E3%83%80%E3%82%A4%E3%82%AA%E3%83%BC%E3%83%89%E3%81%AE%E7%94%9F%E5%AD%98%E3%83%81%E3%82%A7%E3%83%83%E3%82%AF

**Reply by m-tky** [reply]
ありがとうございます。

ご指摘いただいたダイオードを試してみると、反応しなかったため、新しいものに交換し、無事動くようになりました、、！

長い間ありがとうございました。
これからも大事に末長く使います。

**Reply by Ryo624** [reply]
無事に解決していただきましてありがとうございます！
お手数おかけしまして申し訳ありませんでした。

末永くご愛用していただけますと幸いです。
